### PR TITLE
Variable creation

### PR DESCRIPTION
**Pull request by jindrahelcl**
- All used tf.Variable calls were replaced by tf.get_variable.. Some are still left (e.g. in cnn_encoder), but I'd rather leave the decision of how to initialize them to someone who wrote the code. (@jlibovicky?)
- the variables in the attention are now initialized accordingly to the Bahdanau paper
- SentenceEncoder and Decoder now use OrthoGRU cells - a plain GRU cells in a variable scope which by default initializes its variables as random orthogonal matrices.
- And also, this PR shrinks the number of parameters in the test/small testcase

### CODE REVIEW

**Comment by jlibovicky** [comment]
How do you say a variable is regularizable or not?

**Comment by jindrahelcl** [comment]
That's not a part of this pull request..

Anyway, when calling `tf.get_variable`, you could specify a `regularizer` argument, which is a `(Tensor -> Tensor or None)` function.. The resulting Tensor of this function applied on the variable is then stored in the `GraphKeys.REGULARIZATION_LOSSES` collection.

In the trainer, you can get all the variables from this collection by doing something like this
```
tensors = tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES)
l2_loss = sum(tensors)
```

The regularizer function then can be for example `tf.nn.l2_loss`

**Comment by jlibovicky** [comment]
Asi bych upřednostnil rebase and merge před vytvářením mergovacího commitu, budeme aspoň mít přehlednější ten gitovej graf.

**Comment by jlibovicky** [comment]
Někde tam vyleze gradient jako `NaN`, proto to padá na historgram_summaries, ale i kdyby to na nich nespadlo, tak bude loss stejně `NaN`. Nevíš, kde to mohlo vzniknout?

**Comment by jindrahelcl** [comment]
rebasy jsou zapovězený jakmile něco pushneš na net :-)

**Comment by jindrahelcl** [comment]
Nany: To je divný, já to tim pustil u sebe a šlo to.. Sice se to nic neučilo, ale šlo to..

**Comment by martinpopel** [comment]
> rebasy jsou zapovězený jakmile něco pushneš na net :-)

Ne nutně. Záleží, zda na nich někdo založil nějakou další práci.
Třeba @choko rebasuje pull requesty pomocí
https://blog.spreedly.com/2014/06/24/merge-pull-request-considered-harmful/

**Comment by jindrahelcl** [comment]
Je fakt, že je nás málo.. Klidně to rebasujme, mě se taky líbí přehlednější commit history.. Napřed se ale musí vyřešit, že to je blbě. :D
